src/App.jsx

```diff
@@ -1,18 +1,32 @@
 import reactLogo from "./assets/react.svg";
 import viteLogo from "/vite.svg";
 import "./App.css";
-import { useState } from "react"; 
-import TodoList from "./TodoList";
+import { useState, useEffect } from "react";
+import TodoList from "./TodoList.jsx";
 import AddTodoForm from "./AddTodoForm";
 
+function useSemiPersistentState() {
+  const existingTodo = JSON.parse(localStorage.getItem("savedTodoList")) ?? [];
```

Comment:
Nice use of ?? here

For more info on the nullish coalescing operator, you can read this mdn article

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing